Leadlly/leadlly.in
# Fix invalid DOM property warnings (kebab-case to camelCase in SVG)

---
name: Bug Fix Pull Request
about: Submit a pull request to fix a bug in the project
title: 'Fix Invalid DOM Property Warnings for SVG Elements'
labels: 'bug'
assignees: '  '

---

**Issue Number**
Closes #13

**Bug Description**  
There were warnings in the console related to invalid DOM properties where kebab-case attributes (e.g., `stop-color`, `stroke-width`) were used instead of camelCase (`stopColor`, `strokeWidth`). The PR fixes these warnings by converting the attributes to camelCase.

**Video and Screenshot References**

- **Before (Bug):**  
 <img width="1280" alt="Screenshot 2024-10-07 at 10 16 11 AM" src="https://github.com/user-attachments/assets/0db3377f-93ee-4666-9e6c-f97ad91ff62a">

- **After (Fix):**  
 <img width="1392" alt="Screenshot 2024-10-07 at 9 40 59 PM" src="https://github.com/user-attachments/assets/b7b88186-69dc-4d9e-b240-dfc6f1c567e1">

**Testing Steps**  
1. Clone the repository and switch to this `fix-invalid-dom-properties` branch.
2. Run `npm run dev`.
3. Open the browser and ensure that there are no warnings in the console for SVG DOM properties.
4. Verify that the UI works as expected and the SVG icons (such as Logo.tsx) are rendered correctly.

**Checklist**
- [x] I have thoroughly tested the fix.
- [x] I have attached video and picture references.
- [x] The related issue number is attached.
- [x] The bug is resolved, and no new errors are introduced.
- [x] I have checked for potential regressions.

**Additional Context**  

This issue impacted components such as `Logo.tsx` and others where SVG elements were used. The proposed changes resolve the invalid DOM warnings without affecting the design.
